components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -360,9 +362,19 @@ private OAuth2AccessTokenRespDTO generateNewAccessToken(OAuthTokenReqMessageCont
         AccessTokenDO newTokenBean = createNewTokenBean(tokReqMsgCtx, oAuthAppBean, existingTokenBean, timestamp,
                 validityPeriodInMillis, oauthTokenIssuer);
         setDetailsToMessageContext(tokReqMsgCtx, validityPeriodInMillis, newTokenBean, timestamp);
-        // Persist the access token in database
-        persistAccessTokenInDB(tokReqMsgCtx, existingTokenBean, newTokenBean, timestamp,
-                newTokenBean.getAccessToken());
+
+        /* Check whether the existing token needs to be expired and send the corresponding parameters
+        to the persistAccessTokenInDB method. */
+        if (expireExistingToken) {
+            // Persist the access token in database and mark the existing token as expired
+            persistAccessTokenInDB(tokReqMsgCtx, existingTokenBean, newTokenBean, timestamp,
+                    newTokenBean.getAccessToken());
+        } else {
+            // Persist the access token in database without updating the existingTokenBean
```

Comment:
lets the change comment to 
```
// Persist the access token in database without updating the existing token.
// The existing token should already be updated by this point.
```
We need to say that `expireExistingToken == false` should only be when the existing token is already updated in the db to a non active state.

Reply:
Also lets add a debug log to check which condition gets through.